oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/ClusterNodeInfoTest.java

```diff
@@ -417,6 +526,29 @@ public void skipExpiredClusterIdWithDifferentInstanceId() throws Exception {
         info2.dispose();
     }
 
+    // OAK-9564
+    @Test
+    public void cannotGetClusterWithRuntimeId() {
+        ClusterNodeInfo info = newClusterNodeInfo(0);
+        int id = info.getId();
+        assertEquals(1, id);
+        // shut it down
+        info.dispose();
+
+        // edit the runtime ID
+        UpdateOp op = new UpdateOp(String.valueOf(id), false);
+        op.set(ClusterNodeInfo.RUNTIME_ID_KEY, "some-different-uuid");
+        assertNotNull(store.findAndUpdate(Collection.CLUSTER_NODES, op));
+
+        // shouldn't be able to acquire it again
+        try {
+            info = newClusterNodeInfo(0);
```

Comment:
Should rather try to get the specific entry with the id.
```suggestion
            info = newClusterNodeInfo(id);
```